python/pyspark/pandas/data_type_ops/num_ops.py

```diff
@@ -371,15 +384,7 @@ def isnull(self, index_ops: IndexOpsLike) -> IndexOpsLike:
 
     def astype(self, index_ops: IndexOpsLike, dtype: Union[str, type, Dtype]) -> IndexOpsLike:
         dtype, spark_type = pandas_on_spark_type(dtype)
-
-        if isinstance(dtype, CategoricalDtype):
-            return _as_categorical_type(index_ops, dtype, spark_type)
-        elif isinstance(spark_type, BooleanType):
-            return _as_bool_type(index_ops, dtype)
-        elif isinstance(spark_type, StringType):
-            return _as_string_type(index_ops, dtype, null_str=str(np.nan))
-        else:
-            return _as_other_type(index_ops, dtype, spark_type)
+        return _non_fractional_astype(index_ops, dtype, spark_type)
```

Comment:
Don't we need to check `index_ops.hasnans` here?

Reply:
`hasnans` of Decimal may not work as expected as https://issues.apache.org/jira/browse/SPARK-36230.
A TODO comment is added. How do you think about that?

Reply:
I updated the PR/ticket title and description to make that clear.